lib/sgpres.gi

```diff
@@ -3630,49 +3632,49 @@ local DATA,rels,i,j,w,f,r,s,fam,ri,a,offset,rset,re,stack,pres,
     TzSearch(pres);
     TzOptions(pres).lengthLimit:=pres!.tietze[TZ_TOTAL]+1;
   fi;
-  #TzGoGo(pres);
   TzOptions(pres).eliminationsLimit:=5;
-  TzGoElim(pres,subnum);
+  TzGoElim(pres,subnum,wordefs);
   if IsEvenInt(parameter) and Length(GeneratorsOfPresentation(pres))>subnum then
-    warn:=true;
-    # Help Tietze with elimination
-    bad:=Reversed(List(GeneratorsOfPresentation(pres)
-          {[subnum+1..Length(GeneratorsOfPresentation(pres))]},
-          x->LetterRepAssocWord(x)[1]));
-    for i in bad do
-      r:=DATA.secondary[i];
-      re:=true;
-      while re do
-        s:=[];
-        re:=false;
-        for j in r do
-          if AbsInt(j)>subnum then
-            re:=true;
-            if j>0 then
-              Append(s,DATA.secondary[j]);
-            else
-              Append(s,-Reversed(DATA.secondary[-j]));
-            fi;
-          else
-            Add(s,j);
-          fi;
-        od;
-        Info(InfoFpGroup,2,"Length =",Length(s));
-        r:=s;
-        if warn and Length(s)>100*Sum(rels,Length) then
-          warn:=false;
-          Error(
-            "Trying to eliminate all auxiliary generators might cause the\n",
-            "size of the presentation to explode. Proceed at risk!");
-        fi;
-      od;
-      r:=AssocWordByLetterRep(fam,Concatenation(r,[-i]));
-      AddRelator(pres,r);
-      #TzSearch(pres); Do *not* search, as this might kill the relator we
-      #just added.
-      TzEliminate(pres,i);
-    od;
-    Assert(0,Length(GeneratorsOfPresentation(pres))=subnum);
+    Error("did not eliminate properly");
+#    warn:=true;
+#    # Help Tietze with elimination
```

Comment:
What's going on here? A lot of code is commented out and replaced with an `Error`.

Reply:
The MTC uses temporary ("secondary") generators that need to be eliminated with Tietze transformations. So far this was done with a "general" tietze run. That could go wrong and cause situations in which generators get eliminated in the wrong order, not allowing for full generator elimination. The code commented out tried to rescue this.

The change should make this case not needed. Thus the new error message and the eliminated code.
I prefer to initially comment out code, rather than delete it, in case one needs to go back a step. (Yes, one could do it in git, but that is harder.)

Reply:
That makes sense, thank you for the explanation (and sorry for the long delay but I was on vacation)